app/repositories/s3_enrolment_repo.py

```diff
@@ -0,0 +1,107 @@
+import json
+from typing import Any, List
+
+import boto3
+
+from app.config import settings
+from app.domain.entities.enrolment_request import (
+    VALID_STATE_CHANGES,
+    EnrolmentFilters,
+    EnrolmentRequest,
+)
+from app.repositories.enrolment_repo import EnrolmentRepo
+from app.utils.error_handling import handle_s3_errors
+
+
+class S3EnrolmentRepo(EnrolmentRepo):
+    s3: Any
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.s3 = boto3.client("s3", **settings.s3_configuration)
+
+    def get_enrolments(
```

Comment:
get_enrolment_requests is better

Reply:
I think it's better to call it `get_enrolment_authorisations`. It will be easier to understand.